src/histolab/data/__init__.py

```diff
@@ -0,0 +1,264 @@
+        >>> fname = "test-file-for-hash.txt"
+        >>> with open(fname, "w") as f:
+        ...     __ = f.write("content of the file")
+        >>> print(file_hash(fname))
+        0fc74468e6a9a829f103d069aeb2bb4f8646bad58bf146bb0e3379b759ec4a00
+        >>> import os
+        >>> os.remove(fname)
+        """
+        import hashlib
+
+        if alg not in hashlib.algorithms_available:
+            raise ValueError("Algorithm '{}' not available in hashlib".format(alg))
+        # Calculate the hash in chunks to avoid overloading the memory
+        chunksize = 65536
+        hasher = hashlib.new(alg)
+        with open(fname, "rb") as fin:
+            buff = fin.read(chunksize)
+            while buff:
+                hasher.update(buff)
+                buff = fin.read(chunksize)
+        return hasher.hexdigest()
+
+
+def create_image_fetcher():
+    try:
+        import pooch
+    except ImportError:
+        # Without pooch, fallback on the standard data directory
+        # which for now, includes a few limited data samples
+        return None, legacy_data_dir
+
+    pooch_version = __version__.replace(".dev", "+")
+    url = "https://github.com/MPBA/histolab/raw/{version}/histolab/"
+
+    # Create a new friend to manage your sample data storage
+    image_fetcher = pooch.create(
+        # Pooch uses appdirs to select an appropriate directory for the cache
+        # on each platform.
+        # https://github.com/ActiveState/appdirs
+        # On linux this converges to
+        # '$HOME/.cache/histolab-image'
+        # With a version qualifier
+        path=pooch.os_cache("histolab-image"),
+        base_url=url,
+        version=pooch_version,
+        env="HISTOLAB_DATADIR",
+        registry=registry,
+        urls=registry_urls,
+    )
+
+    data_dir = os.path.join(str(image_fetcher.abspath), "data")
+    return image_fetcher, data_dir
+
+
+image_fetcher, data_dir = create_image_fetcher()
+
+if image_fetcher is None:
+    has_pooch = False
+else:
+    has_pooch = True
+
+
+def _has_hash(path, expected_hash):
+    """Check if the provided path has the expected hash."""
+    if not os.path.exists(path):
+        return False
+    return file_hash(path) == expected_hash
+
+
+def _fetch(data_filename):
+    """Fetch a given data file from either the local cache or the repository.
+    This function provides the path location of the data file given
+    its name in the histolab repository.
+    Parameters
+    ----------
+    data_filename:
+        Name of the file in the histolab repository. e.g.
+        'breast/sample1.svs'.
+    Returns
+    -------
+    Path of the local file as a python string.
+    Raises
+    ------
+    KeyError:
+        If the filename is not known to the histolab distribution.
+    ModuleNotFoundError:
+        If the filename is known to the histolab distribution but pooch
+        is not installed.
+    ConnectionError:
+        If histolab is unable to connect to the internet but the
```

Comment:
I'd say:

If the dataset has not been downloaded yet and histolab is unable to connect to the internet